language-server/javascript/src/makeCucumberInfo.ts

```diff
@@ -0,0 +1,62 @@
+import { spawn } from 'child_process'
+import { pipeline, Writable } from 'stream'
+import { NdjsonToMessageStream } from '@cucumber/message-streams'
+import { CucumberInfoStream } from './CucumberInfoStream'
+import { StepDocument } from '@cucumber/suggest'
+import { Expression } from '@cucumber/cucumber-expressions'
+
+export type CucumberInfo = {
+  stepDocuments: readonly StepDocument[]
+  expressions: readonly Expression[]
+}
+
+export function makeCucumberInfo(command: string, args: string[]): Promise<CucumberInfo | null> {
+  const cucumber = spawn(command, args)
+
+  const stderr: Buffer[] = []
+
+  return new Promise((resolve, reject) => {
+    cucumber.on('error', reject)
+    cucumber.on('close', (code, signal) => {
+      if (signal) {
+        reject(new Error(`Received signal ${signal}`))
+      }
+      // https://github.com/cucumber/cucumber-js/issues/1768
```

Comment:
Hopefully we can address this with the [new API](https://github.com/cucumber/cucumber-js/issues/1711) anyway (I'm finally starting to work on that now).

Interested in thoughts about - in addition to a `runCucumber` that does the full test run lifecycle - exposing a few more targeted functions that can help with things like this.